### DISCUSSION TITLE
User email verification

### DISCUSSION THREAD

**Discussion by norkans7** [nyaruka/rapidpro#4912] [Ideas]
We can support email verification for our users form the platform in the following steps

1) Add boolean and charfield for verification status and secret respectively to UserSettings (https://github.com/nyaruka/rapidpro/blob/fc777a112a333851f46482461fd378e26779e791/temba/orgs/models.py#L331)
2) View to turn the boolean field to verified for the passed secret, IMO requires user logged in
3) Add views for users to start the process and send themselves a verification email
4) Add banner for unverified user to encourage them to verify the email for their login

Additionally we could also go further as below later
5) Start requiring verification for views such as exports
6) New signups get redirected to the the verification first
7) Require verification to use any feature once most users have been verified

**Comment by norkans7** [comment]
The boolean defaults to False and CharField as empty until the step 3 when a user can start the process that first set the secret field and once the click the link the view in step 2 will make the boolean to True(as email verified)

**Comment by norkans7** [comment]
https://github.com/nyaruka/rapidpro/pull/4938#discussion_r1377663818

support using different state for the verification status field such  UNVERIFIED, VERIFIED, FAILING

**Comment by rowanseymour** [comment]
We added this .. but now have moved to allauth which is enforcing email verification for all users
